src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -159,15 +166,77 @@ public void FindActivityFunction(SyntaxNodeAnalysisContext context)
                     return;
                 }
 
+                SyntaxNodeUtils.TryGetITypeSymbol(semanticModel, returnTypeNode, out ITypeSymbol returnType);
+
                 SyntaxNodeUtils.TryGetParameterNodeNextToAttribute(context, attribute, out SyntaxNode parameterNode);
 
+                TryGetDefinitionInputType(semanticModel, parameterNode, out ITypeSymbol inputType);
+
                 availableFunctions.Add(new ActivityFunctionDefinition
                 {
                     FunctionName = functionName,
                     ParameterNode = parameterNode,
-                    ReturnTypeNode = returnTypeNode
+                    InputType = inputType,
+                    ReturnTypeNode = returnTypeNode,
+                    ReturnType = returnType
                 });
             }
         }
+
+        private static bool TryGetDefinitionInputType(SemanticModel semanticModel, SyntaxNode parameterNode, out ITypeSymbol definitionInputType)
+        {
+            if (SyntaxNodeUtils.TryGetITypeSymbol(semanticModel, parameterNode, out definitionInputType))
+            {
+                if (SyntaxNodeUtils.IsDurableActivityContext(definitionInputType))
+                {
+                    return TryGetInputTypeFromContext(semanticModel, parameterNode, out definitionInputType);
+                }
+
+                return true;
+            }
+
+            definitionInputType = null;
+            return false;
+        }
+
+        private static bool TryGetInputTypeFromContext(SemanticModel semanticModel, SyntaxNode node, out ITypeSymbol definitionInputType)
+        {
+            if (TryGetDurableActivityContextExpression(semanticModel, node, out SyntaxNode durableContextExpression))
+            {
+                if (SyntaxNodeUtils.TryGetTypeArgumentIdentifier((MemberAccessExpressionSyntax)durableContextExpression, out SyntaxNode typeArgument))
+                {
+                    return SyntaxNodeUtils.TryGetITypeSymbol(semanticModel, typeArgument, out definitionInputType);
+                }
+            }
+
+            definitionInputType = null;
+            return false;
+        }
+
+        private static bool TryGetDurableActivityContextExpression(SemanticModel semanticModel, SyntaxNode node, out SyntaxNode durableContextExpression)
+        {
+            if (SyntaxNodeUtils.TryGetMethodDeclaration(node, out SyntaxNode methodDeclaration))
+            {
+                var memberAccessExpressionList = methodDeclaration.DescendantNodes().Where(x => x.IsKind(SyntaxKind.SimpleMemberAccessExpression));
+                foreach (var memberAccessExpression in memberAccessExpressionList)
+                {
+                    var identifierName = memberAccessExpression.ChildNodes().Where(x => x.IsKind(SyntaxKind.IdentifierName)).FirstOrDefault();
```

Comment:
small nit: `.Where(predicate).FirstOrDefault()` can be replaced with `FirstOrDefault(predicate)`. 

Not a huge problem by any means, just keeps your linq queries a bit more brief.